tests/unit/bases/test_buildd.py

```diff
@@ -855,3 +856,84 @@ def test_warmup_never_network(fake_process, fake_executor, mock_load):
 
     with pytest.raises(BaseConfigurationError):
         base_config.warmup(executor=fake_executor, timeout=0.01, retry_wait=0.1)
+
+
+@pytest.mark.parametrize(
+    "hostname",
+    [
+        "t",
+        "test",
+        "test1",
+        "test-1",
+        "1-test",
+        "this-is-40-characters-xxxxxxxxxxxxxxxxxx",
+        "this-is-63-characters-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+    ],
+)
+def test_set_hostname_unchanged(hostname, logs):
+    """Verify hostnames that are already compliant are not changed."""
+    base_config = buildd.BuilddBase(
+        alias=buildd.BuilddBaseAlias.JAMMY,
+        hostname=hostname,
+    )
+
+    assert base_config.hostname == hostname
+    assert Exact(f"Using hostname '{hostname}'") in logs.debug
+
+
+@pytest.mark.parametrize(
+    "hostname, expected_hostname",
+    [
+        # trim away invalid beginning characters
+        ("-test", "test"),
+        # trim away invalid ending characters
+        ("test-", "test"),
+        ("test--", "test"),
+        ("test1-", "test1"),
+        # trim away invalid characters
+        ("test$", "test"),
+        ("test-!@#$%^&*()test", "test-test"),
+        ("$1test", "1test"),
+        ("test-$", "test"),
+        # this name contains invalid characters so it gets converted, even
+        # though it is 63 characters
+        (
+            "this-is-63-characters-with-invalid-characters-$$$xxxxxxxxxxxxxX",
+            "this-is-63-characters-with-invalid-characters-xxxxxxxxxxxxxX",
+        ),
+        # this name is longer than 63 characters, so it gets converted
+        (
+            "this-is-70-characters-with-valid-characters-xxxxxxxxxxxxxxxxxxXxxxxxxx",
```

Comment:
This test could have the smallest size that gets trimmed (64) to ensure we're trimming at the right length.

Reply:
Good idea, added!